packages/ui-components/src/components/labels-dropdown/labels-dropdown.tsx

```diff
@@ -0,0 +1,196 @@
+import { Component, Element, Event, EventEmitter, Host, Prop, State, Watch, h } from '@stencil/core';
+import { ELabelVisibility, ILabelsDropdown, ILabelsDropdownEvents } from './labels-dropdown.types';
+import { ISelectMultiOptions } from '../select-multi-options/select-multi-options.types';
+import { getFlattenedOptionsMap, getTextFieldConfig } from './labels-dropdown.helper';
+import { CONTAINER_LEFT_MARGIN } from './labels-dropdown.config';
+import { isEmpty, isEqual, isNil } from 'lodash';
+
+@Component({
+	tag: 'kv-labels-dropdown',
+	styleUrl: 'labels-dropdown.scss',
+	shadow: false
+})
+export class KvLabelsDropdown implements ILabelsDropdown, ILabelsDropdownEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) isOpen?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) options?: ISelectMultiOptions = {};
+	/** @inheritdoc */
+	@Prop({ reflect: true }) selectedOptions?: Record<string, boolean> = {};
+	/** @inheritdoc */
+	@Prop({ reflect: true }) filteredOptions?: ISelectMultiOptions;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) noDataAvailableLabel?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) searchable?: boolean = true;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) searchValue?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) minHeight?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) maxHeight?: string;
+
+	@State() _isOpen = this.isOpen;
+	@State() _searchValue = this.searchValue;
+	@State() _overflowingLabelsCounter: number = 0;
+
+	@Watch('isOpen')
+	isOpenWatcher(newValue: boolean, oldValue: boolean) {
+		if (newValue === oldValue) return;
+		this._isOpen = newValue;
+	}
+
+	/** @inheritdoc */
+	@Event() optionsSelected: EventEmitter<Record<string, boolean>>;
+	/** @inheritdoc */
+	@Event() searchChange: EventEmitter<string>;
+	/** @inheritdoc */
+	@Event() selectAll: EventEmitter<void>;
+	/** @inheritdoc */
+	@Event() selectionCleared: EventEmitter<void>;
+
+	@Element() el: HTMLKvLabelsDropdownElement;
+
+	@Watch('options')
+	optionsWatcher(newValue: Record<string, boolean>, oldValue: Record<string, boolean>) {
+		if (isEqual(newValue, oldValue)) return;
+		this.flattenedOptionsHash = getFlattenedOptionsMap(this.options);
+	}
+
+	@Watch('searchValue')
+	searchValueWatcher(newValue: string, oldValue: string) {
+		if (newValue === oldValue) return;
+		this._searchValue = newValue;
+	}
+
+	private dropdownElRef: HTMLKvDropdownElement;
+	private flattenedOptionsHash = getFlattenedOptionsMap(this.options);
+
+	private setOverflowingLabels = (): void => {
+		const containerWidth = this.dropdownElRef.querySelector('.selected-labels-container').clientWidth - CONTAINER_LEFT_MARGIN;
```

Comment:
I believe we should avoid having CSS mixed with JS. But if there's really no other way at least we must use CSS properties to have these values always synced.